src/components/input/index.tsx

```diff
@@ -51,8 +51,10 @@ export class SmoothlyInput {
 				value: this.newState({ value: this.formatter.toString(value), selection: this.state.selection }).value,
 			}
 		}
-		if (value != before)
+		if (value != before) {
+			value = value.trim()
```

Comment:
Have you checked that this does not Interfere with any input that adds in spaces, e.g. PAN, expires etc.

Reply:
`trim()` only removes white space from both ends of a string, not the space in the middle of string and I think most types of input dont even allow you input space.